openlibrary/plugins/upstream/forms.py

```diff
@@ -95,12 +97,18 @@ class RegisterForm(Form):
             help=_("Letters and numbers only please, and at least 3 characters."),
             autocapitalize="off",
             validators=[vlogin, username_validator],
+            pattern="[a-zA-Z0-9]{3,20}",
+            title=_("Between 3 and 20 letters and numbers"),
```

Comment:
To avoid the new string here:

```suggestion
            title=_("Must be between 3 and 20 characters"),
```